### ISSUE TITLE
Inconsistent Usage message needs a thorough clean up

### ISSUE THREAD

**Issue by Delicates** [acmesh-official/acme.sh#3009]
The Usage message in the latest acme.sh (v2.8.7) does not align to the common Usage message conventions and is inconsistent even within itself.
This creates user confusion, struggling to interpret what parameters need to be provided and whether they are optional or mandatory.

### Inconsistency Examples:
acme.sh parameters that require a file a mandatory path argument are described in 3 different ways:
- Argument without brackets: `--webroot, -w  /path/to/webroot`
- Argument in square brackets: `--log    [/path/to/logfile]`
- No argument shown at all: `--cert-file`, `--key-file`, `--ca-file`, `--fullchain-file`

acme.sh parameters that require a mandatory argument from a mutually exclusive options list are described in 2 different ways:
- Options list without brackets: `--notify-level  0|1|2|3`
- Options list in square brackets: `--syslog [0|3|6|7]`

### Convention
Usage messages typically have format convention that can be summarised as follows:
1. Short Single-letter options start with one dash: `-a`
2. Long multi-letter options start with two dashes: `--foo`
3. A single dash with multiple letters usually means the union of those individual short single-letter options rather than a long multi-letter option: `-aAbBcC` is the same as `-a -A -b -B -c -C`
4. Anything in angle brackets < > means a placeholder description for a mandatory argument: `--keylength <bits>`
5. Anything in square brackets [ ] means a placeholder description for an optional argument that can be omitted: `--dnssleep [seconds]`
6. Placeholders should be descriptive: `--keylength <bits>` instead of `--keylength <2048>` or `--keylength <n>`
7. Options separated by a pipe | means those are the valid mutually exclusive values: `0|1|2|3`
8. Ellipsis indicates that additional arguments may be added at this point: `…`
9. Short options should be listed before long options: `-k, --keylength <bits>` instead of `--keylength, -k <bits>`

**Note:** Currently acme.sh parameters `-ccsr`, `-ak` and `-se` are in breach of this convention, so might need to think whether it is worth changing them or not.

### Standard
For more info you can refer to the latest POSIX standard version POSIX.1-2017  (IEEE Std 1003.1-2017) outlined in [The Open Group Base Specifications Issue 7, 2018 edition](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html).

**Comment by Neilpang** [comment]
Thanks, this can be an enhancement in future.

**Comment by Neilpang** [comment]
>> Any interest in me cleaning that up as well?

Yes, but it seems to me not too valuable.  There are a lot of famous software that don't 100% obey the short/long form  rule.
`openssl` is an example.

To me, the most important thing about the command-line form is: readable, usable, and memorable.


>>  I'd simply silently drop the now 'hidden' options, 

Never do that. It's never a good idea to completely drop hidden options.  One of the most important rules is to keep compatible.
There are a lot of users, and there are a lot of blogs and articles about the usage of acme.sh.   We must do our best to keep it working for early users.   No one wants surprises.   And no one wants to learn a tool twice.   One of our goals is to save the user's time.

Once a feature/option is released to the public,  we MUST support it for the whole lifetime.


Yes, people like tidy, clean, and uniform usages/messages,  but the more they like is saving time, readability, and easy-to-use.

These are also the key rules that acme.sh will never drop.

**Comment by lcts** [comment]
I see your point, though personally I'd be OK with changing the API, as long as it is done gracefully with proper warnings etc. Anyway, I'll proceed with the purely cosmetic PR.

Right now, the usage message nearly consistently shows lowercase+hyphenated-unless-adjective/descriptor. In the interest of memorability it might be useful to add lowercase+hyphenated alternate forms for the few long options that don't already follow this pattern, i.e. 

* --createCSR, add --create-csr
* --signcsr, add --sign-csr
* --toPkcs(8), add --to-pkcs(8)
* --stopRenewOnError, add --stop-renew-on-error

and update the usage() accordingly.

**Comment by Neilpang** [comment]
>> Right now, the usage message nearly consistently shows lowercase+hyphenated-unless-adjective/descriptor.

Yes, go ahead.

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Fixed in c0fbe823/d81369d6/2e87e64b (2020-08): the usage message was reformatted for consistency and hyphenated option aliases were added.
